### ISSUE TITLE
Unwanted warning from XS BOOT

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#9281]
Migrated from [rt.perl.org#52572](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=52572) (status was 'resolved')

Searchable as RT52572$

**Comment by p5pRT** [comment]
### From @sisyphus

#### Created by @sisyphus



With perl\-5\.10\.0 \(and later\)\, the following demo script produces the 
warning&#8203;:
Name "main&#8203;::x" used only once&#8203;: possible typo at C&#8203;:/perl/lib/DynaLoader\.pm 
line 224\.

This happens only if global warnings \($^W\) are turned on\. Since the perl 
code in the demo script makes no mention of $x\, I'm wondering whether this 
warning should be produced at all\. \(The fact that the warning specifies line 
224 of DynaLoader\.pm is also a little misleading\.\)

Here's the demo&#8203;:

use Inline C => Config =>
&nbsp;    BOOT => 'sv\_setiv\(get\_sv\("main&#8203;::x"\, 1\)\, 17\);';

use Inline C=> \<\<'EOC';

void greet\(\) \{
&nbsp;    printf\("Hello World\\n"\);
\}

EOC

greet\(\);

This behaviour becomes annoying when\, for example\, one runs 'make test' for 
PDL \(where $PDL&#8203;::SHARE has been created as per the method demo'd above\, in 
case it is ever needed\) and sees the following warning issued for every test 
file that is run&#8203;:

Name "PDL&#8203;::SHARE" used only once&#8203;: possible typo at C&#8203;:/perl/lib/DynaLoader\.pm 
line 226\.

Is this something that should be addressed from within the perl source\, or 
should we be looking at amending the PDL source ?

Cheers\,
Rob

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl 5.10.0:

Configured by Rob at Sat Feb  9 11:01:04 2008.

Summary of my perl5 (revision 5 version 10 subversion 0) configuration:
  Platform:
    osname=MSWin32, osvers=6.0, archname=MSWin32-x64-multi-thread
    uname=''
    config_args='undef'
    hint=recommended, useposix=true, d_sigaction=undef
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags 
' -s -O2 -DWIN32 -DHAVE_DES_FCRYPT -D__NO_ISOCEXT  -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS 
 -fno-strict-aliasing -DPERL_MSVCRT_READFIX',
    optimize='-s -O2',
    cppflags='-DWIN32'
    ccversion='', gccversion='3.4.5', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=undef, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='long long', 
lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='g++', ldflags 
='-s -L"c:\perl510_M\5.10.0\lib\CORE" -L"C:\home\rob\mingw_vista\i686-pc-mingw32\lib"'
    libpth=C:\home\rob\mingw_vista\i686-pc-mingw32\lib
    libs=-lmsvcrt -lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 
 -ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr -lwinmm 
 -lversion -lodbc32 -lodbccp32
    perllibs=-lmsvcrt -lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 
 -ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lws2_32 -lmpr -lwinmm 
 -lversion -lodbc32 -lodbccp32
    libc=-lmsvcrt, so=dll, useshrplib=true, libperl=libperl510.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_win32.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', 
lddlflags='-mdll -s -L"c:\perl510_M\5.10.0\lib\CORE" -L"C:\home\rob\mingw_vista\i686-pc-mingw32\lib"'

Locally applied patches:



@INC for perl 5.10.0:
    C:/perl510_M/5.10.0/lib
    C:/perl510_M/site/5.10.0/lib
    .


Environment for perl 5.10.0:
    HOME (unset)
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=C:\Program Files\Windows 
NT\Accessories;C:\_32\third_party_app;C:\GNUWin32\bin;C:\perl510_M\5.10.0\bin;C:\_32\dmake;C:\home\rob\mingw_vista\i686-pc-mingw32\bin;c:\ruby\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\batch;C:\_32\msys\1.0\bin
    PERL_BADLANG (unset)
    SHELL (unset)


```
</details>

**Comment by p5pRT** [comment]
### From robertmay@cpan.org

On 08/04/2008\, via RT Sisyphus \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> \# New Ticket Created by  "Sisyphus"
>  \# Please include the string&#8203;:  \[perl \#52572\]
>  \# in the subject line of all future correspondence about this issue\.
>  \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=52572 >

>  use Inline C => Config =>
>     BOOT => 'sv\_setiv\(get\_sv\("main&#8203;::x"\, 1\)\, 17\);';

\.\.\.

>  Name "PDL&#8203;::SHARE" used only once&#8203;: possible typo at C&#8203;:/perl/lib/DynaLoader\.pm
>  line 226\.

Look up GV\_ADDMULTI in perldoc perlguts\, and try this&#8203;:

 use Inline C => Config =>
&nbsp;    BOOT => 'sv\_setiv\(get\_sv\("main&#8203;::x"\, TRUE | GV\_ADDMULTI\)\, 17\);';

Regards\,
Rob\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @sisyphus


\-\-\-\-\- Original Message \-\-\-\-\- 
From&#8203;: "Robert May" \<robertmay@&#8203;cpan\.org>
To&#8203;: \<perl5\-porters@&#8203;perl\.org>
Cc&#8203;: \<bugs\-bitbucket@&#8203;netlabs\.develooper\.com>
Sent&#8203;: Tuesday\, April 08\, 2008 6&#8203;:38 PM
Subject&#8203;: Re&#8203;: \[perl \#52572\] Unwanted warning from XS BOOT


> On 08/04/2008\, via RT Sisyphus \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
>> \# New Ticket Created by  "Sisyphus"
>>  \# Please include the string&#8203;:  \[perl \#52572\]
>>  \# in the subject line of all future correspondence about this issue\.
>>  \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=52572 >
>
>>  use Inline C => Config =>
>>     BOOT => 'sv\_setiv\(get\_sv\("main&#8203;::x"\, 1\)\, 17\);';
>
> \.\.\.
>
>>  Name "PDL&#8203;::SHARE" used only once&#8203;: possible typo at 
>> C&#8203;:/perl/lib/DynaLoader\.pm
>>  line 226\.
>
> Look up GV\_ADDMULTI in perldoc perlguts\, and try this&#8203;:
>
> use Inline C => Config =>
>    BOOT => 'sv\_setiv\(get\_sv\("main&#8203;::x"\, TRUE | GV\_ADDMULTI\)\, 17\);';
>

Yep \- that'll do nicely\. \(The bug report can be closed\, as far as I'm 
concerned\.\)

Thanks Rob\.

Cheers\,
Rob

**Comment by p5pRT** [comment]
@jandubois - Status changed from 'open' to 'resolved'
